Logger/FullName.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Logger;
+
+// FullName is a reference type because we do not want the data to be immutable, as a person's
+// name can change (e.g., when they get married). As a record class, the data is not immutable.
+public record class FullName(string First, string Last, string? Middle = null)
+{
+    public void Deconstruct(out string First, out string Last, out string? Middle)
+    {
+        First = this.First;
+        Last = this.Last;
+        Middle = this.Middle;
+    }
+    public override string ToString()
+    {
+        return $"{First} {((Middle!=null)?Middle + " ":"")}{Last}";
```

Comment:
much more simple then the option shown in class. Nice!
```suggestion
        return $"{First} {((Middle!=null)?Middle + " ":"")}{Last}";
   can be written as
        return FirstName + $"{(MiddleName != null ? " " + MiddleName + " " : " ")}" + LastName;
```

Reply:
thanks!